src/Aws/CloudSearch/CloudSearch.php

```diff
@@ -40,13 +40,16 @@ public function upload(Document ...$documents): void
     /**
      * @throws JsonException
      */
-    public function delete(DocumentIdentifier $documentIdentifier): void
+    public function delete(DocumentIdentifier ...$documentIdentifiers): void
     {
         $this->uploadDocuments(
-            [
-                'type' => 'delete',
-                'id' => $documentIdentifier->toString(),
-            ]
+            array_map(
+                fn (DocumentIdentifier $documentIdentifier) => [
```

Comment:
Tutaj można dać:
```suggestion
               static fn (DocumentIdentifier $documentIdentifier) => [
```
Dzięki temu podczas procesu funkcja jest singleton'em w zakresie całego wątku i gdyby doszło do multiplikacji klasy, to zawsze będzie tylko jedna instancja funkcji, bez względu na ilość obiektów.

Reply:
Fajna uwaga, nie myślałem o tym, dzięki za komętarz :)